feat: LEAP-1110: Improve events and update docs for events

Gondragos opened the pull request:
- update docs about new added events
- fix doc about `selectAnnotation` event
- add useful information for `selectHistory` and `beforeSaveAnnotation` events

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [x] Not sure (briefly explain the situation below)

It could but we didn't use arguments from `selectHistory` yet.

### Which logical domain(s) does this change affect?
`Docs`, `Events`

Gondragos committed:
feat: LEAP-1110: Make events improvements

Gondragos committed:
docs: LEAP-1110: Update docs for events

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9369487198

Gondragos committed:
Add missed event payload

Gondragos committed:
Merge branch 'fb-leap-1110/cs-polishing' of ssh://github.com/heartexlabs/label-studio into fb-leap-1110/cs-polishing

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9371014247

hlomzik commented on docs/source/guide/frontend_reference.md:
```diff
@@ -285,6 +285,18 @@ Draft is sent to the server.
+Returning `false` from this event will prevent saving the annotation.
+
+| Argument         | Type     | Description                                                                                                              |
+|------------------|----------|--------------------------------------------------------------------------------------------------------------------------|
+| `labelStudio`    | `Object` | Instance of Label Studio                                                                                                 |
+| `annotation`     | `Object` | Current annotation                                                                                                       |
+| `payload`        | `Object` | Additional information                                                                                                   |
+| `payload.event`  | `string` | Indicates which event is going to be next (`submitAnnotation`, `updateAnnotation`) |
```
```suggestion
|------------------|----------|---------------------------------------------------------------------------------------------|
| `labelStudio`    | `Object` | Instance of Label Studio                                                                                                 |
| `annotation`     | `Object` | Current annotation                                                                                                       |
| `payload`        | `Object` | Additional information                                                                                                   |
| `payload.event`  | `string` | Indicates which event is about to be executed (`submitAnnotation`, `updateAnnotation`, etc.) |
```

Commit:
Make payload.event description better

Co-authored-by: hlomzik <hlomzik@gmail.com>

Gondragos committed:
Remove unnecessary spaces

Gondragos committed:
Merge branch 'fb-leap-1110/cs-polishing' of ssh://github.com/heartexlabs/label-studio into fb-leap-1110/cs-polishing

# Conflicts:
#	docs/source/guide/frontend_reference.md

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9388439156)
> Successfully merged:  10 files changed, 115 insertions(+), 15 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-leap-1110/cs-polishing'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9388439156

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9388460674

caitlinwheeless commented on docs/source/guide/frontend_reference.md:
```diff
@@ -307,10 +319,13 @@ Annotation is updated.
+|-----------------------|-----------|-----------------------------------------------|
+| `annotation`          | `Object`  | Current annotation                            |
+| `previousAnnotation`  | `Object`  | Previous annotation                           |
+| `payload`             | `Object?` | Additional information                        |
```
Check question marke - and could we just say "Payload content" or something under Description

caitlinwheeless replied:
```suggestion
| `payload`             | `Object` | Payload content                        |
```
